server/src/storage/compressed_data_manager.rs

```diff
@@ -0,0 +1,204 @@
+/* Copyright 2022 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Support for managing all compressed data that is inserted into the [`StorageEngine`].
+
+use std::collections::{HashMap, VecDeque};
+use std::path::PathBuf;
+
+use datafusion::arrow::record_batch::RecordBatch;
+use tracing::{info, info_span};
+
+use crate::storage::time_series::CompressedTimeSeries;
+
+/// Signed integer since compressed data is inserted first and the remaining bytes are checked after.
+/// This means that the remaining bytes can be negative briefly until compressed data is saved to disk.
+const COMPRESSED_RESERVED_MEMORY_IN_BYTES: isize = 5000;
+
+/// Stores data points compressed as models in memory to batch compressed data before saving it to
+/// Apache Parquet files.
+pub struct CompressedDataManager {
+    /// Path to the folder containing all compressed data managed by the [`StorageEngine`].
+    data_folder_path: PathBuf,
```

Comment:
It is my understanding that the difference between `PathBuf` and `Path` is the same as `String` and `&str`. Since we want the struct to own the field, we use `PathBuf`. Since `PathBuf` is used extensively, it might be odd to document the difference only here. The alternative would be to document it in all places where it is used but this also seems problematic.

For now I have not made this change but if you think that it is still a good idea, I can do it. Also note that this won't matter if we switch to `object_store` later.

Reply:
I was mostly thinking of `PathBuf` as a mutable `Path` and was not aware that it is unsized. But since `Path` is an unsized type I agree that `PathBuf` should be used until we switch to `object_store`.